pkg/providers/cloudstack/validator.go

```diff
@@ -141,14 +142,17 @@ func (v *Validator) ValidateClusterMachineConfigs(ctx context.Context, cloudStac
 		return fmt.Errorf("validating controlPlaneConfiguration.Endpoint.Host: %v", err)
 	}
 
-	for _, machineConfig := range cloudStackClusterSpec.machineConfigsLookup {
+	for _, machineConfig := range cloudStackClusterSpec.CloudStackMachineConfigs {
 		if len(machineConfig.Spec.Users) <= 0 {
 			machineConfig.Spec.Users = []anywherev1.UserConfiguration{{}}
 		}
 		if len(machineConfig.Spec.Users[0].SshAuthorizedKeys) <= 0 {
 			machineConfig.Spec.Users[0].SshAuthorizedKeys = []string{""}
 		}
-		if err = v.validateMachineConfig(ctx, cloudStackClusterSpec.datacenterConfig, machineConfig); err != nil {
+		if machineConfig.Spec.Users[0].Name == "" {
+			machineConfig.Spec.Users[0].Name = v1alpha1.DefaultCloudStackUser
+		}
```

Comment:
If there were no users specified, the logic here would default the SSH key for the default user but not the username and there would be an error late during the creation process. This adds defaulting logic for the username as well to avoid that.

Reply:
Could you check with @mitalipaygude on how webhook would validate the ssh key, only the ssh key itself or include user?

Reply:
Ok I'll follow up.  For reference, this is the error that occurs if the username is not specified, despite the CLI auto generating:

`Error: applying capi spec: executing apply: error: error validating "STDIN": error validating data: ValidationError(KubeadmControlPlane.spec.kubeadmConfigSpec.users[0]): missing required field "name" in io.x-k8s.cluster.controlplane.v1beta1.KubeadmControlPlane.spec.kubeadmConfigSpec.users; if you choose to ignore these errors, turn validation off with --validate=false`

Reply:
Confirmed with @mitalipaygude. The web hook would validate the ssh and default the user. The same is done in all the other providers.